src/perfrecord.cpp

```diff
@@ -71,10 +75,51 @@ static QStringList sudoOptions(const QString& sudoBinary)
     return options;
 }
 
+static bool privsAlreadyElevated()
+{
+    struct ValueMap {
```

Comment:
while it's ok, I do think it's pretty overkill as we only have these two values to check... it's more code to make it this generic over just calling the lambda twice, no?